modules/references/server/controllers/reference.server.controller.js

```diff
@@ -0,0 +1,221 @@
+'use strict';
+
+var mongoose = require('mongoose'),
+    _ = require('lodash'),
+    path = require('path'),
+    async = require('async'),
+    errorService = require(path.resolve('./modules/core/server/services/error.server.service')),
+    emailService = require(path.resolve('./modules/core/server/services/email.server.service')),
+    pushService = require(path.resolve('./modules/core/server/services/push.server.service')),
+    Reference = mongoose.model('Reference'),
+    User = mongoose.model('User');
+
+/**
+ * Validate the request body and data consistency
+ * of Create a reference
+ */
+function validateCreate(req) {
+  var valid = true;
+  var details = {};
+  var interactionErrors = {};
+
+  // Can't create a reference to oneself
+  if (req.user._id.toString() === req.body.userTo) {
+    valid = false;
+    details.userTo = 'self';
+  }
+
+  // Some interaction must have happened
+  var isInteraction = req.body.interactions && (req.body.interactions.met || req.body.interactions.hostedMe || req.body.interactions.hostedThem);
+  if (!isInteraction) {
+    valid = false;
+    interactionErrors.any = 'missing';
+  }
+
+  // Value of 'recommend' must be valid ('yes', 'no', 'unknown')
+  if (req.body.recommend && !['yes', 'no', 'unknown'].includes(req.body.recommend)) {
```

Comment:
Hehe, `includes` is pretty new JS but since this is backend and we're at the brink of moving to ES6, let's roll with it. :thumbsup:

Reply:
:smile: Indeed. This drops support for [node 5 and less](https://node.green/#ES2016-features-Array-prototype-includes).
Hopefully this will be replaced with some validation library in the future. Reinventing the wheel here. :)

Reply:
Yap, let's add validation library to the todo and loop back to it later. Would you mind opening a quick issue about it and add it to backlog at project board I just made: https://github.com/Trustroots/trustroots/projects/4 ?

Reply:
#807